### PR TITLE
Add context to 'too many segments' error in decoder.

### PR DESCRIPTION
**Pull request by lthibault**
Adds maximum observed segment ID and max segments for the given stream to error returned by decoder.

### CODE REVIEW

File: message.go [hunk 1]
```diff
@@ -688,7 +689,8 @@ func (d *Decoder) Decode() (*Message, error) {
 	}
 	maxSeg := SegmentID(binary.LittleEndian.Uint32(d.wordbuf[:]))
 	if maxSeg > maxStreamSegments {
-		return nil, errors.New("decode: too many segments to decode")
+		return nil, fmt.Errorf("decode: too many segments (%d) to decode (max=%d)",
```
Comment (zenhack) [inline message.go:692]: Per #364, can we void fmt here and use strconv instead?

Comment (lthibault) [reply message.go:692]: Sure.  Should I define a separate error type?

Comment (zenhack) [reply message.go:692]: You don't need to; you could also just concatenate strings and keep using errors.New. Your call.

Comment (zenhack) [reply message.go:692]: (You probably want to use internal/str.Utod instead of strconv directly)

**Review by zenhack — APPROVED** [review]

**Pull request merged**